Formula/slither-analyzer.rb

```diff
@@ -0,0 +1,56 @@
+class SlitherAnalyzer < Formula
+  include Language::Python::Virtualenv
+
+  desc "Solidity static analysis framework written in Python 3"
+  homepage "https://github.com/crytic/slither"
+  url "https://github.com/crytic/slither/archive/refs/tags/0.8.3.tar.gz"
+  sha256 "7549ab983541094a34525d7e74af1fa9b1538122c92a6ed969d2e4c7fcc7b42b"
+  license "AGPL-3.0-only"
+  head "https://github.com/crytic/slither.git", branch: "master"
+
+  depends_on "python@3.10"
+  depends_on "solc-select"
+
+  resource "crytic-compile" do
+    url "https://github.com/crytic/crytic-compile/archive/refs/tags/0.2.3.tar.gz"
+    sha256 "b41cc3ca6e43b70c2257dba5c1e794ca0881577f908acee9bc5bdfb132792e30"
+  end
```

Comment:
We already have a `crytic-compile` formula -- can't we use that?

Reply:
Hi @carlocab! crytic-compile is both an standalone program that you can run, as well as a library that you can import and use. In this case, Slither imports and calls things from crytic-compile ([example](https://github.com/crytic/slither/blob/ce9dbf650d7acfee51ddafd844929f4e8d345672/slither/core/slither_core.py#L13-L14))

My understanding is that each of these formulas creates a separate python venv, so importing across them wouldn't work. I may be missing something though, I'm pretty new to brew packaging. How do you normally handle these?

Reply:
You can write a `.pth` file into this formula's private `site-packages` directory that will allow it to access `crytic-compile`'s Python libraries.

See `ansible-lint` for an example (though the syntax is a little outdated):

https://github.com/Homebrew/homebrew-core/blob/3836ab56271f54f57f08923e00d8a18ffc112c4a/Formula/ansible-lint.rb#L175-L181

Reply:
Awesome, thanks! I've now added crytic-compile as a dependency, and write the pth during this formula's installation to add the crytic-compile library